crates/models/src/catalogs.rs

```diff
@@ -61,6 +61,17 @@ impl Catalog {
         let generator = schemars::gen::SchemaGenerator::new(settings);
         generator.into_root_schema_for::<Self>()
     }
+
+    /// Returns the names of all specs that are directly included within this catalog.
+    /// This does _not_ include specs from imported catalogs.
+    pub fn all_spec_names(&self) -> impl Iterator<Item = &str> {
+        self.collections
+            .keys()
+            .map(AsRef::<str>::as_ref)
```

Comment:
Fancy!

Reply:
lol after our conversation yesterday, I just had to try it the other way :upside_down_face: